fastestimator/trace/metric/auc.py

```diff
@@ -0,0 +1,149 @@
+                Calculate metrics for each label, and find their average, weighted
+                by support (the number of true instances for each label).
+            ``'samples'``:
+                Calculate metrics for each instance, and find their average.
+            Will be ignored when ``y_true`` is binary.
+        sample_weight: array-like of shape (n_samples,), default=None. Sample weights.
+        max_fpr: float > 0 and <= 1, default=None
+            If not ``None``, the standardized partial AUC over the range [0, max_fpr] is returned.
+            For the multiclass case, ``max_fpr``, should be either equal to ``None`` or ``1.0`` as AUC ROC partial
+            computation currently is not supported for multiclass.
+        multi_class : {'raise', 'ovr', 'ovo'}, default='raise'
+            Only used for multiclass targets. Determines the type of configuration to use. The default value raises
+            an error, so either
+            ``'ovr'`` or ``'ovo'`` must be passed explicitly.
+            ``'ovr'``:
+                Stands for One-vs-rest. Computes the AUC of each class against the rest. This
+                treats the multiclass case in the same way as the multilabel case. Sensitive to class imbalance even
+                when ``average == 'macro'``, because class imbalance affects the composition of each of the
+                'rest' groupings.
+            ``'ovo'``:
+                Stands for One-vs-one. Computes the average AUC of all possible pairwise combinations of classes.
+                Insensitive to class imbalance when ``average == 'macro'``.
+        labels : array-like of shape (n_classes,), default=None
+            Only used for multiclass targets. List of labels that index the classes in ``y_score``. If ``None``,
+            the numerical or lexicographical order of the labels in ``y_true`` is used.
+        per_ds: Whether to automatically compute this metric individually for every ds_id it runs on, in addition to
+            computing an aggregate across all ds_ids on which it runs. This is automatically False if `output_name`
+            contains a "|" character.
+
+        Raises:
+            ValueError: Expected values of 'max_fpr' should be either None or between 0 and 1.
+            ValueError: Expected values of 'multi_class' are ['raise', 'ovr' or 'ovo'].
+            ValueError: Expected values of 'average' should be either None or ['micro', 'macro', 'samples', 'weighted'].
+    """
+    def __init__(self,
+                 true_key: str,
+                 pred_key: str,
+                 mode: Union[None, str, Iterable[str]] = ("eval", "test"),
+                 ds_id: Union[None, str, Iterable[str]] = None,
+                 output_name: str = "auc",
+                 average: str = "macro",
+                 sample_weight: List = None,
+                 max_fpr: float = None,
+                 multi_class: str = 'raise',
+                 labels: List = None,
```

Comment:
```suggestion
                 labels: Optional[List[str]] = None,
```

Reply:
Is this a list of str or some other type?

Reply:
its an array of shape (num_of_samples)

Reply:
Right, but the question is about the contents of the array, not its shape